cpp/include/graph.hpp

```diff
@@ -214,5 +214,182 @@ class GraphCSC: public GraphCompressedSparseBase<VT,ET,WT> {
   {}
 };
 
+/**
+ * @brief       Base class for graphs constructed in the C++ API
+ *
+ * This class will own edge data, until the data is moved.
+ *
+ * @tparam VT   Type of vertex id
+ * @tparam ET   Type of edge id
+ * @tparam WT   Type of weight
+ */
+template <typename VT, typename ET, typename WT>
```

Comment:
I think this is a good time to decide between VT, ET, WT and vertex_t, edge_t, weight_t (or value_t). May start a slack discussion and a quick vote?

Reply:
Sure, I'm fine with having a discussion and making a decision.

Reply:
Sure, I would prefer VT, etc. We will have to type these a lot and I would like to type less.

Reply:
> Sure, I would prefer VT, etc. We will have to type these a lot and I would like to type less.

The number one goal should be code readability. In large long running projects, people spend more time on reading code than writing, so I think "typing less" is not as important as code readability.

Reply:
Agree about the goal of readability.

I try to use short variable names and template type names only in short blocks of code where you won't lose the context of what the variable means (e.g. short loops, index variables in a lambda, templates where the type is truly generic).